### PR TITLE
[Feature]Added models views and serializers in cooperativeand autho

### PR DESCRIPTION
**Pull request by arj9912**

### CODE REVIEW

File: autho/serializers.py [hunk 1]
```diff
@@ -1,17 +1,30 @@
...
             "citizenship_issued_place",
             "citizenship_issued_date",
             "dob",
-            "father_name",
+            "fathers_name",
             "phone_number",
         ]
+
+    def create(self, validated_data):
+
```
Comment (RahulRimal) [inline autho/serializers.py:27]: remove this line

File: common/helpers.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+
```
Comment (RahulRimal) [inline common/helpers.py:1]: remove these two lines

File: common/mixins.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+from shortuuidfield import ShortUUIDField
+
+from rest_framework import serializers
```
Comment (RahulRimal) [inline common/mixins.py:3]: put this import at top

File: common/mixins.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+
+class BaseModelSerializerMixin(serializers.ModelSerializer):
+    # idx = serializers.CharField(read_only=True)
```
Comment (RahulRimal) [inline common/mixins.py:7]: remove this

File: common/mixins.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    idx = serializers.ReadOnlyField()
```
Comment (RahulRimal) [inline common/mixins.py:8]: remove this

File: common/mixins.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+                representation[field_name] = [
+                    getattr(related_instance, "idx", related_instance.id)
+                    for related_instance in related_instances
+                ]
+        return representation
+
+
+class BaseApiMixin():
+    lookup_field = "idx"
+
```
Comment (RahulRimal) [inline common/mixins.py:40]: only one line

File: cooperative/models.py [hunk 7]
```diff
@@ -42,19 +47,14 @@ class Loan(BaseModelMixin):
     emi_amount = models.DecimalField(max_digits=10, decimal_places=2)
     currently_outstanding = models.DecimalField(max_digits=12, decimal_places=2)
     total_due = models.DecimalField(max_digits=12, decimal_places=2)
-    personal_guarantors = models.ManyToManyField(
-        PersonalGuarantor, related_name="guaranted_loans", blank=True
-    )
-
+    # personal_guarantors = models.ManyToManyField(
```
Comment (RahulRimal) [inline cooperative/models.py:50]: remove commented code

**Review by RahulRimal — CHANGES_REQUESTED** [review]

File: autho/serializers.py (hunk 1 above)
Comment (arj9912) [reply autho/serializers.py]: ok done

File: common/helpers.py (hunk 2 above)
Comment (arj9912) [reply common/helpers.py]: ok done

File: common/mixins.py (hunk 3 above)
Comment (arj9912) [reply common/mixins.py:3]: ok done

File: common/mixins.py (hunk 4 above)
Comment (arj9912) [reply common/mixins.py]: ok done

File: common/mixins.py (hunk 5 above)
Comment (arj9912) [reply common/mixins.py]: ok done

File: common/mixins.py (hunk 6 above)
Comment (arj9912) [reply common/mixins.py:37]: ok done

File: cooperative/models.py (hunk 7 above)
Comment (arj9912) [reply cooperative/models.py]: ok done

**Review by RahulRimal — APPROVED** [review]

**Pull request merged**